packages/event-handler/src/rest/BaseRouter.ts

```diff
@@ -133,11 +139,49 @@ abstract class BaseRouter {
     };
   }
 
-  public abstract resolve(
+  public async resolve(
     event: unknown,
     context: Context,
     options?: ResolveOptions
-  ): Promise<unknown>;
+  ): Promise<APIGatewayProxyResult | undefined> {
+    if (!isAPIGatewayProxyEvent(event)) {
+      this.logger.warn(
+        'Received an event that is not compatible with this resolver'
+      );
+      return;
+    }
+
+    try {
+      const request = proxyEventToWebRequest(event);
+      const path = new URL(request.url).pathname as Path;
```

Comment:
In general, I dislike casting, should these two assignments  use type guards?

Reply:
I dislike it too.

My rule of thumb is that if this is a TypeScript type system limitation (aka TS can't detect the type even though it should) or we're reasonably sure that this type is the only realistic code path, then type casting can be ok.

If instead there's a runtime scenario in which the value might not be of that type, then a type guard is preferred.

Reply:
For the line below, `as HttpMethod`, should also be fine since it comes directly from the request we just parsed.

Reply:
I've created a type guard for the HTTP method but I've moved the check outside the `try` block because if we throw a `MethodNotAllowedError`, inside the `catch` block, the error handling logic will try to create the `Request` object and throw with the `TypeError` mentioned above. Rather than throw a `MethodNotAllowedError`, we just return a `APIGatewayProxyResult`  with the appropriate 405 error code.

The creation of the Wed API `Request` object has also been moved out of the `catch` block. If this fails it is an unrecoverable error so we should just let the lambda fail.